app/controllers/grade_entry_forms_controller.rb

```diff
@@ -47,6 +47,23 @@ def view_summary
     @date = params[:date]
   end
 
+  # returns the column distribution/breakdown for each grade_entry_item in a grade_entry_form
+  #   (see the second chart in the summary of a marks spreadsheet, accessed from the dashboard)
+  def column_breakdown
+    grade_entry_form = GradeEntryForm.find(params[:id])
+    grade_entry_items = grade_entry_form.grade_entry_items
+    return_data = { labels: [], datasets: [] }
+    axis_labels = (0..100).step(5).to_a
+    dict_data = []
+    grade_entry_items.each do |item|
```

Comment:
Use `.map` here, returns an array, so that you don't need to manually add elements to `dict_data`.